<a href="https://colab.research.google.com/github/DinaShaim/GB_ML_MathStat/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D1%8F%D1%8F_%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%B0_%D0%9C%D0%B0%D1%82%D1%81%D1%82%D0%B0%D1%82_%D0%A3%D1%80%D0%BE%D0%BA_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашняя работа 5
### Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

**Задание 1** 

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

**Решение**



Ковариация равна математическому ожиданию произведения отклонений случайных величин:

$$cov_{XY} = M(XY) - M(X)M(Y)$$


In [3]:
import numpy as np

In [7]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp

array([ 35,  45, 190, 200,  40,  70,  54, 150, 120, 110])

In [8]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks

array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [9]:
cov = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov

9157.839999999997

In [13]:
np.cov(zp, ks, ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Зная ковариацию и среднее квадратичное отклонение каждого из двух признаков, можно вычислить коэффициент корреляции Пирсона:

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [14]:
r = cov / (np.std(zp)*np.std(ks))
r

0.8874900920739158

In [16]:
r1 = np.corrcoef(zp,ks)
r1

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

**Задание 2** 

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

**Решение**

Обозначим известные данные:

In [17]:
import pandas as pd
df = pd.DataFrame({'IQ' : [131, 125, 115, 122, 131, 115, 107, 99, 125, 111] })
X = df['IQ'].mean()
X

118.1

In [18]:
σ = df['IQ'].std(ddof=1)
σ

10.54566788359614

Обьем выборки равен 10, значит степеней свободы будет 9 и значение критерия возьмем из таблицы, тогда:

Вычислим доверительный интервал по формуле:

$$T_{1,2} = \overline{X} \pm  t_{a/2}\cdot \frac{σ}{\sqrt{n}}  $$

In [19]:
T1 = X - 2.262*(σ/np.sqrt(10))
T1

110.55660776308164

In [20]:
T2 = X + 2.262*(σ/np.sqrt(10))
T2

125.64339223691834

Доверительный интервал равен [110.55 ; 125.64]

**Задание 3** 

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

**Решение**

Вычислим доверительный интервал по формуле:

$$T_{1,2} = \overline{X} \pm  Z_{a/2}\cdot \frac{σ}{\sqrt{n}}  $$

$$T_{1,2} = 80 \pm  1.96\cdot \frac{16}{16}  $$

Доверительнвй интерал равен: 

$$ [78.04; 81.96]$$

In [21]:
X = 174.2
D = 25
n = 27
σ = np.sqrt(D)

In [22]:
T1 = X - 1.96*(σ/np.sqrt(n))
T1

172.31398912064722

In [23]:
T2 = X + 1.96*(σ/np.sqrt(n))
T2

176.08601087935276

Доверительнвй интерал равен: 

$ [172.31; 176.08]$